File: DecompiledGameScripts/core/components/scriptComponents/vendorComponent.swift

public class VendorComponent extends ScriptableComponent {

  @runtimeProperty("customEditor", "TweakDBGroupInheritance;Vendors.VendingMachine")
  private edit let m_vendorTweakID: TweakDBID;

  private edit const let m_junkItemArray: [JunkItemRecord];

  @runtimeProperty("customEditor", "AudioEvent")
  @default(VendorComponent, dev_vending_machine_processing)
  private edit let m_brandProcessingSFX: CName;

  @runtimeProperty("customEditor", "AudioEvent")
  @default(VendorComponent, dev_vending_machine_can_falls)
  private edit let m_itemFallSFX: CName;

  public final const func GetVendorID() -> TweakDBID {
    return this.m_vendorTweakID;
  }

  public final const func GetJunkItemIDs() -> [JunkItemRecord] {
    return this.m_junkItemArray;
  }

  public final const func GetJunkCount() -> Int32 {
    return ArraySize(this.m_junkItemArray);
  }

  public final const func GetProcessingSFX() -> CName {
    return this.m_brandProcessingSFX;
  }

  public final const func GetItemFallSFX() -> CName {
    return this.m_itemFallSFX;
  }
}

File: Definitions/Game_Definitions/classes/VendorComponent.lua
---@meta
---@diagnostic disable

---@class VendorComponent : gameScriptableComponent
---@field vendorTweakID TweakDBID
---@field junkItemArray JunkItemRecord[]
---@field brandProcessingSFX CName
---@field itemFallSFX CName
VendorComponent = {}

---@return VendorComponent
function VendorComponent.new() return end

---@param props table
---@return VendorComponent
function VendorComponent.new(props) return end

---@return CName
function VendorComponent:GetItemFallSFX() return end

---@return Int32
function VendorComponent:GetJunkCount() return end

---@return JunkItemRecord[]
function VendorComponent:GetJunkItemIDs() return end

---@return CName
function VendorComponent:GetProcessingSFX() return end

---@return TweakDBID
function VendorComponent:GetVendorID() return end

